packages/alea-frontend/components/ProblemList.tsx

```diff
@@ -0,0 +1,258 @@
+  const currentFilepath = node.filepath || parentFilepath;
+
+  if (node.title === '' || node.title) {
+    const cleanTitle = removeHtmlTags(node.title);
+    result.push({
+      title: cleanTitle,
+      archive: currentArchive,
+      filepath: currentFilepath,
+      id: node.id,
+      level: level,
+    });
+  }
+
+  if (Array.isArray(node.children)) {
+    node.children.forEach((child) => {
+      result = result.concat(
+        extractTitlesAndMetadata(child, level + 1, currentArchive, currentFilepath)
+      );
+    });
+  }
+
+  return result;
+};
+
+const ProblemList: FC<ProblemListProps> = ({ data, courseId }) => {
+  const [problemCounts, setProblemCounts] = useState<Record<string, number>>({});
+  const router = useRouter();
+  const { home } = getLocaleObject(router);
+  const t = home.courseThumb;
+
+  useEffect(() => {
+    if (!courseId) return;
+
+    axios
+      .get(`/api/get-course-problem-counts/${courseId}`)
+      .then((resp) => {
+        setProblemCounts(resp.data);
+      })
+      .catch((error) => {
+        console.error('Error fetching problem counts:', error);
+      });
+  }, [courseId]);
+
+  const handleButtonClick = (
+    archive?: string,
+    filepath?: string,
+    title?: string,
+    courseId?: string
+  ) => {
+    router.push({
+      pathname: '/per-section-quiz',
+      query: { archive, filepath, title, courseId },
+    });
+  };
+
+  const titlesAndMetadata = data ? extractTitlesAndMetadata(data) : [];
+
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+        height: '100vh',
+        padding: 2,
+        position: 'relative',
+        backgroundColor: '#f0f4f8',
+      }}
+    >
+      <Typography
+        variant="h3"
+        sx={{
+          mb: 3,
+          textAlign: 'center',
+          color: '#2b3f5c',
+          textTransform: 'uppercase',
+          letterSpacing: '1.5px',
+          textShadow: '2px 2px 4px rgba(0, 0, 0, 0.5)',
+        }}
+      >
+        {t.practice}&nbsp;
+      </Typography>
+      <Paper
+        sx={{
+          padding: 3,
+          borderRadius: 2,
+          boxShadow: 3,
+          width: '100%',
+          maxWidth: 800,
+          maxHeight: '80vh',
+          overflowY: 'auto',
+          textAlign: 'left',
+          backgroundColor: '#ffffff',
+          borderLeft: `5px solid ${PRIMARY_COL}`,
+        }}
+      >
+        <List>
+          {titlesAndMetadata.map((item, index) => {
+            const problemCount = problemCounts[item.id] || 0;
+            const isBold = item.level === 2;
+            const fontWeight = isBold ? 'bold' : 'normal';
+            const backgroundColor = item.level % 2 === 0 ? '#f0f4f8' : 'blue';
+            const borderRadius = item.level % 2 === 0 ? '8px' : '0px';
+            const fontFamily =
+              item.level === 2 ? 'Times New Roman, Times, serif' : 'Roboto, sans-serif';
+
```

Comment:
Can we write a separate  function for this

Reply:
Yeah we can write a separate function but i think in this particular case having related logic inline within map is better and also overhead of calling a function will not be there for every iteration of map on each item.